src/Tizen.Sensor/Tizen.Sensor/SensorRecorder.cs

```diff
@@ -0,0 +1,265 @@
+            {
+                int error = Interop.SensorRecoder.OptionSetInt(option.optionHandle, RecorderOption.Interval, value);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error checking SetInterval");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "OptionBuilder.SetInterval Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder option is not created");
+            }
+        }
+
+        /// <summary>
+        /// gets the OptionHandle object.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public RecordOption getOption()
+        {
+            return option;
+        }
+
+    }
+
+    /// <summary>
+    /// This class provides the option handle.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class RecordOption
+    {
+
+        /// <summary>
+        /// Property: Gets the optionHandle.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <value> The option handle </value>
+        public IntPtr optionHandle { set; get; }
+    }
+
+    /// <summary>
+    /// The SensorRecorder class is used for controlling the sensor recorder(start,stop). 
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class SensorRecorder
+    {
+        private bool _disposed = false;
+        private int sensorType;
+
+        /// <summary>
+        /// Initializes a new instance of the SensorRecorder class.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name='type'>
+        /// type refers to sensor type
+        /// </param>
+        public SensorRecorder(int type)
+        {
+            sensorType = type;
+
+            if (IsSupported())
+            {
+                throw new NotSupportedException(" recorder is not supported.");
+            }
+        }
+
+        /// <summary>
+        /// Checks whether it is supported to record a given sensor type.
+        /// </summary>
+        /// <returns>
+        /// returns true if it is supported to record a given sensor type.
+        /// </returns>
+        /// <since_tizen> 6 </since_tizen>
+        private bool IsSupported()
+        {
+            bool isSupported;
+            int error = Interop.SensorRecoder.IsSupported(sensorType, out isSupported);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error checking if sensor recorder is supported");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.IsSupported Failed");
+            }
+            return isSupported;
+        }
+
+        /// <summary>
+        ///  Starts to record a given sensor type.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void Start(RecordOption option)
```

Comment:
Parameters of methods must have valid documentation. Method has 1 parameter, 0 are documented.